http-common/src/connector.rs

```diff
@@ -83,16 +79,13 @@ impl Incoming {
                         std::sync::Arc::new(tokio::sync::Semaphore::new(MAX_REQUESTS_PER_USER))
                     })
                     .clone();
-                let mut timeout_stream = tokio_io_timeout::TimeoutStream::new(unix_stream);
-                timeout_stream.set_read_timeout(Some(READ_WRITE_TIMEOUT));
-                timeout_stream.set_write_timeout(Some(READ_WRITE_TIMEOUT));
 
                 let server = crate::uid::UidService::new(ucred.uid(), server.clone());
                 tokio::spawn(async move {
                     match user_state.try_acquire_owned() {
                         Ok(_permit) => {
                             if let Err(http_err) = hyper::server::conn::Http::new()
-                                .serve_connection(Box::pin(timeout_stream), server)
+                                .serve_connection(Box::pin(unix_stream), server)
```

Comment:
@arsing I wasn't able to get TimeoutStream (either by changing it to TimeoutReader or only setting the read timeout) to respect the timeouts in IS. From my testing, the READ_WRITE_TIMEOUT (5 sec) always stopped whatever API call I was making and returned an error if that API call was taking longer than 5 sec. So I changed it back to streams that don't use timeouts.

Reply:
Okay. We can revisit it later.